### PR TITLE
[LWD] fix(desktop): prevent RCE via shell.openExternal URL validation

### PR DESCRIPTION
**Pull request by mcayuelas-ledger**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 12 unit tests added for URL validation.
- [x] **Impact of the changes:**
      - Markdown component link handling
      - Platform catalog search URL opening
      - All external URL opening via `openURL()` function

### 📝 Description

**Problem**:The `Markdown` component and other parts of the codebase call `shell.openExternal()` without validating the URL protocol, allowing Remote Code Execution (RCE) through malicious links.

Attack vectors include:
- `file://` URLs to execute local binaries
- `smb://` URLs for NTLM hash theft on Windows
- Application protocol handlers (ms-msdt, zoommtg, etc.)

**Solution**: 
- Added `isUrlSafe()` function in `linking.ts` to validate URL protocols
- Only `http:` and `https:` protocols are allowed
- Updated `Markdown.tsx` and `NoResult.tsx` to use the safe `openURL()` function
- Added 12 comprehensive unit tests covering allowed protocols, blocked protocols, and edge cases

**Files modified:**
- `src/renderer/linking.ts` - Added URL validation
- `src/renderer/linking.test.ts` - Added unit tests (new file)
- `src/renderer/components/Markdown.tsx` - Use safe openURL
- `src/renderer/screens/platform/v2/Catalog/Browse/Result/NoResult.tsx` - Use safe openURL

### 🔒 Security

This fix addresses a RCE vulnerability. The pattern follows Electron security best practices:
- [Electron Security: Do not use openExternal with untrusted content](https://www.electronjs.org/docs/latest/tutorial/security#15-do-not-use-openexternal-with-untrusted-content)
- [CWE-78: OS Command Injection](https://cwe.mitre.org/data/definitions/78.html)

### ❓ Context

- **JIRA or GitHub link**: Security fix -https://ledgerhq.atlassian.net/browse/LIVE-25290

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/linking.ts [hunk 1]
```diff
@@ -1,10 +1,32 @@
 import { track } from "~/renderer/analytics/segment";
 import electron from "electron";
+
 let shell: Electron.Shell | undefined;
 if (!process.env.STORYBOOK_ENV) {
   shell = electron.shell;
 }
+
+const ALLOWED_PROTOCOLS = ["http:", "https:"];
```
Comment (Justkant) [inline apps/ledger-live-desktop/src/renderer/linking.ts]: Are we sure this is not used for deeplinks ?
